### PR TITLE
Fixed dataflow if invoked action returns a future

### LINKED ISSUE #2008: Compilation error in dataflow of action returning a future
On gcc 4.9, boost 1.54, hpx built in release with commit 60fff02b8d2edbc5ed7017ce3073b2735a588592, [this example](https://gist.github.com/ltroska/e65c21462327ea16fe52) fails to compile with [this error](https://gist.github.com/ltroska/e51adaf6d9002c764fd7).

### PR DESCRIPTION
**Pull request by hkaiser**
This fixes #2008

### CODE REVIEW

**Comment by hkaiser** [comment]
@ltroska This should fix your issue, please verify.

**Comment by ltroska** [comment]
@hkaiser The issue is fixed with these changes.

File: tests/regressions/lcos/dataflow_action_2008.cpp [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+hpx::future<double> foo()
+{
+    return hpx::make_ready_future(42.);
+}
+
+HPX_DEFINE_PLAIN_ACTION(foo);
+
+int hpx_main(int argc, char* argv[])
+{
+    hpx::future<hpx::future<double> > f =
```
Comment (sithhell) [inline tests/regressions/lcos/dataflow_action_2008.cpp]: shouldn't this be future<double> instead?

Comment (hkaiser) [inline tests/regressions/lcos/dataflow_action_2008.cpp]: No, this is correct. The actual inconsistency is that `async<Action>()` unwraps any returned future...

**Pull request merged**